pilot/pkg/features/pilot.go

```diff
@@ -218,6 +218,14 @@ var (
 			"outbound listener for each pod in a headless service. This feature should be disabled "+
 			"if headless services have a large number of pods. ",
 	)
+
+	EnableUnsafeRegex = env.RegisterBoolVar(
+		"PILOT_ENABLE_UNSAFE_REGEX",
```

Comment:
are there functional differences between the old and new regex? if not, then why provide a feature flag?

Reply:
Yes, a lot of features of std::regex are not supported in re2, that is why Envoy moved to the safe_regex field to avoid API breakage: https://github.com/envoyproxy/envoy/pull/7878

Reply:
then this should be true by default for this release

Reply:
@rshriram why does the default matter? Either way at some point we have to change it to `true`, what good does having it to false for a release do?

Reply:
this is for unsafe one. not the safe one

Reply:
I think there are two cases.

* We set this to `false` by default. Users might have behavior change if they are using some regex that happens to behave differently. In our release notes, we add a clear notice to users about this. Users that notice this set the flag to true if needed. Likely a small % of users impcted
* We set this to `true` by default. 99% of users will not configure this, so most of them will have the "unsafe regex". Eventually Envoy removes unsafe_regex and we have to set it to false, which gives us the same result as above

So it seems setting it to `true` just pushes the problem back. Am I thinking about this wrong?

Reply:
I think we (or envoy) sholud provide a migration tool. For us if an expression can be expressed with full fidelity with re2, then we can just use `safeRegex`, for others we *should*  keep it as unsafe.

The trickier part is if there is if both re2 and std::regex accept a regex, but behave in different ways.

We can be more aggressive with runtime settings `PILOT_ENABLE_UNSAFE_REGEX=0` if we have such a tool.

Reply:
Is it feasible to provide such a tool? I am not familiar with the differences of the two regex libraries

Reply:
To me it is more about the optics and the ensuing uproar that istio broke backward compatibility. And as you said, we are simply delaying the inevitable. I think we need some more louder broadcast in discuss.istio.io (cc @geeknoid ) and other places to make sure that people are aware of this and they should really know that it is not under our control/i.e. this is a security fix in envoy for everyone's good, yada yada

Reply:
Not sure what the general opinion here is now. IMO we should be pushing to migrate the default setting ASAP, but allow existing users to still set this env variable in 1.4. In future releases, this might not work anymore anyway (because Envoy will have removed unsafe regexes by then), so I don't think setting the default to unsafe makes a lot of sense; it will just shorten the migration period.

Reply:
My opinion:  the move to libc++ addressed the security vulnerability.  This is more about re2 being a more efficient impl than libc++'s std::regex.  So there's no emergency here.  We should default re2 off in 1.4, and announce at that time that we will default re2 on for 1.5.

Reply:
Sounds good to me. We can merge this as "on" (use std::regex) and then next week after the branch is cut flip it to "off" on master

Reply:
Hey @howardjohn just to be sure- we just merged this as "off" (ie safe_regex), so we'll need another PR to switch it to "on" (regex) for the release

Reply:
sorry we had some more discussion outside this PR that didn't make its way here. The consensus we got from a few people was we should move forward with safe regex and make it clear in the upgrade notice. Thanks!!

Reply:
I see, thank you!